Sensor_parameters:
  frequency : 100

  pos_offset : 1.0
  pos_gain : 250.0

  neg_offset : 1.0
  neg_gain : -25.25

  atm_offset : 101.325

  
  Pressure_pos_index : 0
  Pressure_neg_index : 1
  Pressure_macro_index : 2


Reference_parameters:
  frequency : 1000

PWM_parameters:
  frequency : 1000

  pid_pos_index : 18
  pid_neg_index : 19

positive_channel:
  n_channel : 3

negative_channel:
  n_channel : 3

MPC_parameters:
  NP : 5
  n_x : 1
  n_u : 3
  Ts : 0.01
  Q_values : 10
  R_values : 1
  pos_ku_micro : 0.5
  pos_ku_macro : 0.5
  pos_ku_atm : 2
  neg_ku_micro : 3
  neg_ku_macro : 3
  neg_ku_atm : 6
  
channel_volume : # mL (= 1e-6m^3)
  pos1 : 100
  pos2 : 100
  pos3 : 100
  neg1 : 100
  neg2 : 100
  neg3 : 100


PID_parameters:
  n_channel : 3

  pos_PID_gains:
    kp : 1.0
    ki : 2.0
    kd : 3.0

  neg_PID_gains:
    kp : 1.0
    ki : 2.0
    kd : 3.0

system_parameters :
  sensor_print : true
  reference_print : true
  pwm_print : true
  valve_operate : false
